persistent/Database/Persist/Quasi/Internal.hs

```diff
@@ -1392,6 +1392,14 @@ takeForeign ps entityName = takeRefTable
                                 EntityNameHS refTableName
                             , foreignRefTableDBName =
                                 EntityNameDB $ psToDBName ps refTableName
+                            , -- TODO: The existing foreign key syntax for
+                              -- UnboundForeignDef is not sufficiently rich to
+                              -- allow specifying the schema of the foreign
+                              -- relation. We need to add the ability to parse
+                              -- schema=foo directives inline for foreign keys
+                              -- and insert those values here.
```

Comment:
This adds another `.persistentmodels` parser task.

I inferred that we need to make this change based on the block comment on `UnboundForeignDef`:
```haskell
-- | Define an explicit foreign key reference.
--
-- @
-- User
--     name Text
--     email Text
--
--     Primary name email
--
-- Dog
--     ownerName Text
--     ownerEmail Text
--
--     Foreign User fk_dog_user ownerName ownerEmail
-- @
--
-- @since 2.13.0.0
data UnboundForeignDef
```

Reply:
summarized the task in this ticket: https://linear.app/mercury/issue/INT-848/hackday-persistent-extend-parser-to-allow-schemas-for-fkey-references

Reply:
linear ticket: https://linear.app/mercury/issue/INT-848/hackday-persistent-extend-parser-to-allow-schemas-for-fkey-references

Reply:
hopefully resolved with #9